core/templates/pages/topics-and-skills-dashboard-page/navbar/topics-and-skills-dashboard-navbar.directive.ts

```diff
@@ -61,6 +63,10 @@ angular.module('oppia').directive('topicsAndSkillsDashboardNavbar', [
               RubricObjectFactory.create(SKILL_DIFFICULTIES[0], []),
               RubricObjectFactory.create(SKILL_DIFFICULTIES[1], ['']),
               RubricObjectFactory.create(SKILL_DIFFICULTIES[2], [])];
+            // Create an 8-digit random int as the temporary ID for the skill.
+            var randomId = Math.floor(Math.random() * 100000000).toString();
+            ContextService.setCustomEntityContext(
```

Comment:
You might need a new method on ContextService to set a "temp" context -- otherwise, this looks like the context for the actual skill with ID randomId, which seems like a misrepresentation.